src/main/java/org/apache/sling/commons/threads/impl/DefaultThreadPoolManager.java

```diff
@@ -404,23 +426,56 @@ public ThreadPoolExecutor getExecutor() {
             return null;
         }
 
-        protected void unregisterMBean() {
+        protected void unregisterMBeanAndMetrics() {
             if ( this.mbeanRegistration != null ) {
                 this.mbeanRegistration.unregister();
                 this.mbeanRegistration = null;
             }
+            unregisterGauges();
         }
 
-        protected void registerMBean() {
+        protected void registerMBeanAndMetrics() {
             try {
                 final Dictionary<String, String> mbeanProps = new Hashtable<String, String>();
                 mbeanProps.put("jmx.objectname", "org.apache.sling:type=threads,service=ThreadPool,name=" + this.name);
 
                 final ThreadPoolMBeanImpl mbean = new ThreadPoolMBeanImpl(this);
                 this.mbeanRegistration = bundleContext.registerService(ThreadPoolMBean.class.getName(), mbean, mbeanProps);
+
+                // not exposing MaxThreadAge as bean.getMaxThreadAge() is deprecated due
+                // to SLING-6261 and always returns -1
+                String[] ignoredAttributes = { "MaxThreadAge" };
```

Comment:
I think it's clearer if we make this a list outright (you're calling `Arrays.asList` later on anyway) and make it a static field.

Reply:
Done but since it became a constant (static field) also moved it back to ThreadPoolMetricsGauges as it seems closes to the responsible entity.